### PR TITLE
Update npm deps for be runnable with higher node versions

### PR DESCRIPTION
**Pull request by zerdos**
I updated a couple of npm dependencies - the goal was to be able to run the project with node 11.
Also I made sure that it still runs with node 8.2.1

Also, may I ask why the package-locks are not checked in? 
Z.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,74 +45,69 @@
     "test-mocha-snapshot": "mocha scripts/mocha-snapshot-spec.js",
     "check-node-version": "node scripts/check-node-version.js"
   },
-  "lint-staged": {
-    "*.js": [
-      "precommit-lint"
```
Comment (zerdos) [inline package.json]: I moved this down to the file, next to the husky config

File: packages/desktop-gui/src/main.jsx [hunk 2]
```diff
@@ -7,8 +7,11 @@ import handleGlobalErrors from './lib/handle-global-errors'
 import momentOverrides from './lib/configure-moment'
 
 import App from './app/app'
+import {  } from 'mobx';
 
-useStrict(true)
+configure({
+    enforceActions: 'observed'
```
Comment (zerdos) [inline packages/desktop-gui/src/main.jsx]: with the latest mobx version the useStrict became deprecated

File: packages/example/gulpfile.js [hunk 3]
```diff
@@ -5,15 +5,14 @@ let RevAll = require('gulp-rev-all')
 let runSequence = require('run-sequence')
 
 gulp.task('assets', function () {
-  let revAll = new RevAll({
+
+  return gulp.src('./app/**/*')
+  .pipe(RevAll.revision({
```
Comment (zerdos) [inline packages/example/gulpfile.js]: new version of RevAll changed the signiture

**Comment by jennifer-shehane** [comment]
Well, we have sadly neglected our npm deps in this repo. The plan is to have an automatic update system as outlined here: https://github.com/cypress-io/cypress/issues/2644 So, in the meantime of a larger refactor, we have not checked in our package-locks.

**Comment by chrisbreiding** [comment]
Some dependencies cannot be updated or are best not updated. We cannot update `mobx` to 5 in the `runner` because it drops support for IE11, which we're planning on supporting in the future. It's okay because `mobx` 4 is still maintained. I've looked into updating `jsdom` and think it's more trouble than it's worth because of breaking changes and incompatibilities with enzyme. Some updates, like moving to CoffeeScript 2, are breaking changes for users since it's not compatible with CoffeeScript 1, so we have to be very careful with it.

This is a bit much to bite off all at once. As you can see, it's causing tests to fail in every suite we have. I think it might be better to try and tackle one package at a time. That will make it easier to debug failures.

**Comment by jennifer-shehane** [comment]
Yes, I would also recommend 1 npm dependency per PR, so that we can isolate exactly where the failures are occurring. 

@zerdos I will be closing this PR. If you would like, open new PRs and please focus on 1 npm dependency per PR.

**Pull request closed without merging**